Summer

SamuraiLevel100 opened the pull request:
show all info about film

make a page FilmDetail which will take id from url and make call to api

And just style it with mui, use chatGPT for stuff like this
Dont forget about Types for models, it is important

SamuraiLevel100 committed:
Pause

Pause before developing a page adaptation

SamuraiLevel100 committed:
Movie details without using API

Movie details without using API

SamuraiLevel100 committed:
API connection

API connection, formation and display of data about a specific movie, template display of the list of available sessions (data for sessions are taken from the _mocks_ file)

SamuraiLevel100 committed:
Delete folder vite.

-_-

SamuraiLevel100 committed:
Merge branch 'dev' of https://github.com/GentleCats/cinema-fe into summer

Maks262626 commented on src/pages/FilmDetails.tsx:
```diff
@@ -1,5 +1,199 @@
+
+  useEffect(() => {
+    const fetchFilmDetails = async () => {
+      try {
+        const response = await axios.get(`https://localhost:7046/api/Movie/get-all`);
+        if (response.data && Array.isArray(response.data)) {
+          const filmData = response.data.find((f: Film) => f.id === Number(id));
+          if (filmData) {
+            setFilm(filmData);
+            // setFilmSessions(filmData.sessions || []);  
+            const mockSessions = sessions.filter(session => session.film.id === filmData.id);
+
+            setFilmSessions(mockSessions);
+          }
+        } else {
```
use GET method -> get-by-id/{id}

Maks262626 commented on src/pages/FilmDetails.tsx:
```diff
@@ -1,5 +1,199 @@
+
+    fetchFilmDetails();
+  }, [id]);
+
+  if (loading) {
+    return <Typography variant="h5">Loading...</Typography>;
+  }
+
```
use Loader component here

Maks262626 reviewed (changes requested):
Make Session Component,
Make SessionList Component,

SamuraiLevel100 committed:
Making changes from comments

Make Session Component,
Make SessionList Component,
-
use GET method -> get-by-id/{id}
-
use Loader component here
-
changed the display of movie cards in FilmList
